### PR TITLE
RSDK-7375 Update modbus to use new board interface

### PR DESCRIPTION
**Pull request by oliviamiller**
Updating to the new board interface. This includes:

- Analog read returning struct (this wont be in stable until next weeks release so waiting to merge this until then)
- Removing `status`
- `analogreader` renamed to `analog` and `writeanalog` on the `analog` interface
- Addition of `streamTicks`
- `digitalInterruptByName` now returns error instead of bool

### CODE REVIEW

File: modbus_tcp_board/analog_pin.go
Comment (oliviamiller) [inline modbus_tcp_board/analog_pin.go:36]: I don't see a way of determining the range and stepSize of the analog reader this is using, so still just returning the bit value.

**Review by JohnN193 — APPROVED** [review]
LGTM, just want to check if we should be bumping the [version](https://github.com/viam-soleng/viam-modbus/blob/d344041e3b3465a2c76576e7c6bde096d644aafe/utils/utils.go#L3)? we probably want to do a new release of the module

File: modbus_tcp_board/analog_pin.go
Comment (penguinland) [reply modbus_tcp_board/analog_pin.go:36]: Discussion question: how typical is it to not be able to provide this info? Is there other info that could have been provided instead if we'd designed this feature differently (if the scope doc had said something else)? I wonder if it will cause confusion for other people implementing their own board components in the wild.

**Review by penguinland — APPROVED** [review]
LGTM!

Remember to update go.mod next week when the new version of the RDK is out (that's the kind of thing I would forget to do).

File: modbus_tcp_board/analog_pin.go
Comment (JohnN193) [reply modbus_tcp_board/analog_pin.go:36]: i believe its not typical. the issue in modbus's case is we could be talking to any board and we have no way to ask/guess what the info should be. Other boards will have an answer from the datasheet, or (in the revpis case) be configurable

**Comment by oliviamiller** [comment]
> LGTM, just want to check if we should be bumping the [version](https://github.com/viam-soleng/viam-modbus/blob/d344041e3b3465a2c76576e7c6bde096d644aafe/utils/utils.go#L3)? we probably want to do a new release of the module
yeah i will bump version and upload new module version to registry

File: go.mod
Comment (penguinland) [inline go.mod:9]: v0.27.1 was released last week. If we get rid of the hyphen and everything after it, does this PR still work? If so, we don't need to wait until next week's release...

Comment (oliviamiller) [reply go.mod:9]: yep, we need to wait then will i will bump this to v0.28 and merge

Comment (oliviamiller) [reply go.mod:9]: the analog read changes are in next weeks release

**Pull request merged**
